source/fundamentals/linq.txt

```diff
@@ -0,0 +1,531 @@
+   :dedent:
+   :start-after: start-restaurant-model
+   :end-before: end-restaurant-model
+
+.. _csharp-linq-queryable:
+
+Make A Collection Queryable
+---------------------------
+
+To use LINQ with to query your collection you must first make the collection
+queryable. To make your collection queryable, use the ``AsQueryable`` method
+as follows:
+
+.. code-block:: csharp
+   :emphasize-lines: 2
+
+    _restaurantsCollection = db.GetCollection<Restaurant>("restaurants");
+    var queryableCollection = _restaurantsCollection.AsQueryable();
+
+Once the collection is queryable, you can compose a query using 
+**query comprehension syntax** or **method syntax**.
+
+Select the :guilabel:`Query Syntax` or :guilabel:`Method Syntax` tab to see how 
+how to compose a query using LINQ:
+
+.. tabs::
+
+   .. tab:: Query Syntax
+      :tabid: query-syntax
+
+      .. code-block:: csharp
+
+         var query = from r in queryableCollection
+             where r.Name == "The Movable Feast"
+             select new { r.Name, r.Address };
+
+   .. tab:: Method Syntax
+      :tabid: method-syntax
+
+      .. code-block:: csharp
+
+         var query = queryableCollection
+             .Where(r => r.Name == "The Movable Feast")
+             .Select(r => new { r.Name, r.Address });
+
+The documents found in the query are returned as an object that implements the
+`IQueryable <https://learn.microsoft.com/en-us/dotnet/api/system.linq.iqueryable-1?redirectedfrom=MSDN&view=net-6.0>`__
+interface. You can print the results of the preceding example as follows: 
+
+.. io-code-block::
+
+   .. input::
+      :language: csharp
+
+      foreach (var restaurant in query)
+      {
+          Console.WriteLine(restaurant.ToJson());
+      }
+
+   .. output::
+      
+      { "name" : "The Movable Feast", "address" : { "building" : "284", "coord" : [-73.982923900000003, 40.6580753], "street" : "Prospect Park West", "zipcode" : "11215" } }
+
+Supported Aggregation Stages
+----------------------------
+
+You can create an aggregation pipeline by using LINQ with supported aggregation
+operations. To learn more about each aggregation pipeline stage, see the server
+manual :ref:`aggregation-pipeline-operator-reference` page.
+
+$project
+~~~~~~~~
+
+The ``$project`` aggregation stage returns a document using only the specified
+fields.
+
+Select the :guilabel:`Query Syntax` or :guilabel:`Method Syntax` tab to see how 
+how to generate a ``$project`` stage using LINQ:
+
+.. tabs::
+
+   .. tab:: Query Syntax
+      :tabid: query-syntax
+
+      .. code-block:: csharp
+         :emphasize-lines: 2
+
+         var query = from r in queryableCollection
+             select new { r.Name, r.Address };
+
+   .. tab:: Method Syntax
+      :tabid: method-syntax
+
+      .. code-block:: csharp
+         :emphasize-lines: 2
+
+         var query = queryableCollection
+             .Select(r => new { r.Name, r.Address });
+
+The preceding example returns the following document:
```

Comment:
I think I naturally fell into doing it this way because IMO the page is more focused on how to translate that aggregation stage into a LINQ query, not necessarily the outcome of doing it. What the query actually does seems more like secondary information in this case.